### PR TITLE
Clean up the resources during uninstall

### PR DESCRIPTION
**Pull request by bhatpmk**
# Description

This change deletes / removes the resources which are created as part of Verrazzano install, during the uninstallation. Also the changes are made to ensure that Verrazzano uninstall doesn't clean the resources, which were there before the installation.

Fixes VZ-2188

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh [hunk 1]
```diff
@@ -110,10 +121,12 @@ function delete_coherence_operator {
       error "Failed to uninstall the Coherence Kubernetes operator."
     fi
   fi
+  kubectl delete validatingwebhookconfigurations.admissionregistration.k8s.io coherence-operator-validating-webhook-configuration --ignore-not-found
```
Comment (kminder) [inline platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh:124]: I hand't considered this workaround for the issue.  I guess this is fine if it works but we really need to fix the underlying issue also.  Is deleting all the webhooks a general approach we want to take?

File: platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh [hunk 2]
```diff
@@ -60,6 +60,17 @@ function delete_verrazzano() {
   log "Deleting Verrazzano namespaces"
   delete_k8s_resources namespace ":metadata.name,:metadata.labels" "Could not delete Verrazzano namespaces" '/k8s-app:verrazzano.io|verrazzano-system/ {print $1}' \
     || return $? # return on pipefail
+
+  # Clean-up
+  log "Deleting leftover resources"
```
Comment (kminder) [inline platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh:65]: Might want to consider revising this log message but I don't have a better suggestion.

File: platform-operator/scripts/uninstall/uninstall-steps/0-uninstall-applications.sh [hunk 3]
```diff
@@ -62,6 +65,11 @@ function delete_oam_components {
   delete_k8s_resource_from_all_namespaces components.core.oam.dev
 }
 
+# Delete the
+function delete_oam_components {
+  delete_k8s_resource_from_all_namespaces components.core.oam.dev
```
Comment (kminder) [inline platform-operator/scripts/uninstall/uninstall-steps/0-uninstall-applications.sh:70]: This function seems out of place and/or incomplete.

File: platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh (hunk 2 above)
Comment (bhatpmk) [reply platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh:65]: @kminder I will fix this log message

File: platform-operator/scripts/uninstall/uninstall-steps/0-uninstall-applications.sh (hunk 3 above)
Comment (bhatpmk) [reply platform-operator/scripts/uninstall/uninstall-steps/0-uninstall-applications.sh]: The duplicate function was a mistake by me, removed it now.

File: platform-operator/scripts/uninstall/uninstall-utils.sh [hunk 4]
```diff
@@ -56,8 +56,13 @@ function delete_k8s_resource_from_all_namespaces() {
       fi
     done
   fi
+  # Delete the CRD
```
Comment (bobdonat) [inline platform-operator/scripts/uninstall/uninstall-utils.sh:59]: The delete of CRDS seems out of place in this function.  Is there something I am missing here?

Comment (bhatpmk) [reply platform-operator/scripts/uninstall/uninstall-utils.sh]: @bobdonat The existing code fails to remove the CRD, when there is no CR based on that in the cluster. The newly added code deletes the CRD, which is not cleaned-up as part of deleting the operator.

File: platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh [hunk 5]
```diff
@@ -33,22 +33,22 @@ function delete_verrazzano() {
     || return $? # return on pipefail
 
   log "Deleting Verrazzano crds"
-  delete_k8s_resources crds ":metadata.name" "Could not delete CustomResourceDefinitions from Verrazzano" '/verrazzano.io/ && ! /verrazzanos.install.verrazzano.io/' \
+  delete_k8s_resources crds ":metadata.name" "Could not delete CustomResourceDefinitions from Verrazzano" '/verrazzano.io/ && ! /verrazzanos.install.verrazzano.io/ && ! /verrazzanomanagedclusters.clusters.verrazzano.io/' \
     || return $? # return on pipefail
 
   # deleting certificatesigningrequests
```
Comment (bobdonat) [inline platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh:39]: Is this section intended to be commented out?

File: platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh [hunk 6]
```diff
@@ -33,22 +33,22 @@ function delete_verrazzano() {
...
-  log "Deleting CertificateSigningRequests"
-  delete_k8s_resources csr ":metadata.name" "Could not delete CertificateSigningRequests from Verrazzano" '/csr-/' \
-    || return $? # return on pipefail
+  #log "Deleting CertificateSigningRequests"
```
Comment (bobdonat) [inline platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh:40]: Is this section intended to be commented out?

File: platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh (hunk 1 above)
Comment (bhatpmk) [reply platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh:124]: It is working for this case.

File: platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh (hunk 6 above)
Comment (bhatpmk) [reply platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh:40]: With a fresh OKE cluster, I can see the following CertificateSigningRequest -

kubectl get CertificateSigningRequest --show-kind -o custom-columns=NAME:.metadata.name,KIND:.kind --ignore-not-found --no-headers -n default
csr-6kd4q   CertificateSigningRequest
csr-nmt7x   CertificateSigningRequest
csr-wv584   CertificateSigningRequest

I don't think we should delete them during Verrazzano uninstall.

File: platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh [hunk 7]
```diff
@@ -33,22 +33,17 @@ function delete_verrazzano() {
     || return $? # return on pipefail
 
   log "Deleting Verrazzano crds"
-  delete_k8s_resources crds ":metadata.name" "Could not delete CustomResourceDefinitions from Verrazzano" '/verrazzano.io/ && ! /verrazzanos.install.verrazzano.io/' \
-    || return $? # return on pipefail
-
-  # deleting certificatesigningrequests
```
Comment (bobdonat) [inline platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh:39]: Still not clear why deleting CertificateSigningRequests is being removed.  Was it causing a problem?

Comment (bhatpmk) [reply platform-operator/scripts/uninstall/uninstall-steps/3-uninstall-verrazzano.sh:39]: @bobdonat These are the CertificateSigningRequests created as part of the cluster creation, not created by verrazzano installation.

**Review by bobdonat — APPROVED** [review]

**Pull request merged**